### PR TITLE
Add timestamp to WMS request

### PR DESCRIPTION
**Pull request by nhaney90**
A simple application that shows how to apply timestamps to WMS requests to prevent requests from being cached by the browser.

### CODE REVIEW

File: web-js/add-timestamp-to-wms-request/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+});
+```
+
+This is the request sent by the unmodified method:
+
+http://sampleserver1c.arcgisonline.com/ArcGIS/services/Specialty/ESRI_StatesCitiesRivers_USA/MapServer/WMSServer?SERVICE=WMS&REQUEST=GetMap&FORMAT=image/png&TRANSPARENT=TRUE&STYLES=&VERSION=1.3.0&LAYERS=2&WIDTH=1852&HEIGHT=450&CRS=EPSG:102100&BBOX=-15439274.142032105,4568743.987222122,-6379346.053449143,6770130.401834613
+
+This is the request sent by the re-implemented getImageUrl method:
+
+http://sampleserver1c.arcgisonline.com/ArcGIS/services/Specialty/ESRI_StatesCitiesRivers_USA/MapServer/WMSServer?SERVICE=WMS&REQUEST=GetMap&FORMAT=image/png&TRANSPARENT=TRUE&STYLES=&VERSION=1.3.0&LAYERS=2&WIDTH=1852&HEIGHT=450&CRS=EPSG:102100&BBOX=-15439274.142032105,4568743.987222122,-6379346.053449143,6770130.401834613&_timeStamp=1484064028192
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.md]: i think it'd be clearer if you only displayed the additional parameter here.

File: web-js/add-timestamp-to-wms-request/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+				})
+			return result;
+		};
+	}
+});
+```
+
+This is the request sent by the unmodified method:
+
+http://sampleserver1c.arcgisonline.com/ArcGIS/services/Specialty/ESRI_StatesCitiesRivers_USA/MapServer/WMSServer?SERVICE=WMS&REQUEST=GetMap&FORMAT=image/png&TRANSPARENT=TRUE&STYLES=&VERSION=1.3.0&LAYERS=2&WIDTH=1852&HEIGHT=450&CRS=EPSG:102100&BBOX=-15439274.142032105,4568743.987222122,-6379346.053449143,6770130.401834613
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.md]: please break out individual url parameters on their own lines. if you want to keep the hyperlink, maybe do it off the word [request](http://sampleserver1c.arcgisonline.com/ArcGIS/services/Specialty/ESRI_StatesCitiesRivers_USA/MapServer/WMSServer?SERVICE=WMS&REQUEST=GetMap&FORMAT=image/png&TRANSPARENT=TRUE&STYLES=&VERSION=1.3.0&LAYERS=2&WIDTH=1852&HEIGHT=450&CRS=EPSG:102100&BBOX=-15439274.142032105,4568743.987222122,-6379346.053449143,6770130.401834613)

File: web-js/add-timestamp-to-wms-request/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#Add Timestamp to WMS Requests
+
+##About
+Browsers often cache images to boost performance by limiting the number of requests that web applications need to send. While this feature is often helpful, it can be a hinderance in situations where data is constantly being updated. Consider this example. Let's say you are working with a WMS service showing realtime locations of ships. Thus while the extent of the map may not change (triggering no new requests) the data has changed meaning the image needs to be updated. Setting a refresh interval on the WMSLayer is of course the best way to handle this problem. However you will probably find your browser caches the image and does not re-issue the request because the request parameters have not changed. One potential workaround for this issue is to use timestamps.
+
+##Usage Notes
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.md]: be explicit about the fact that a proxy is required.  link to the doc to help folks set it up and explain that its the reason we can't host a live demo.

File: web-js/add-timestamp-to-wms-request/WMSLayerTimeStamp.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+			var wmsLayer = new WMSLayer("http://sampleserver1.arcgisonline.com/ArcGIS/services/Specialty/ESRI_StatesCitiesRivers_USA/MapServer/WMSServer", {
+				format: "png",
+				visibleLayers: [2]
+			});
+			  
+			map.addLayer(wmsLayer);
+			  
+			map.on("layer-add-result", function(evt) {
+				var getImageUrl = wmsLayer.getImageUrl;
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/WMSLayerTimeStamp.html]: please add inline code comments to explain the flow.

File: web-js/add-timestamp-to-wms-request/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#Add Timestamp to WMS Requests
+
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.md]: did you forget to display your .gif from within the README?

Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.md]: nevermind, the .gif in this PR is from the upload shapefile sample.  including another for the WMS sample, would be cool too, but not mandatory.

File: web-js/upload-shapefile-to-agol/index.html [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+					<button type = "submit" class = "btn btn-primary" >Submit</button>
+				</form>
+			</div>
+			<div id="map" class="column">
+				<h1 id="pendingText" class="pending-text">Pending</h1>
+			</div>
+		</div>
+	</div>
+</body>
+</html>
```
Comment (jgravois) [inline web-js/upload-shapefile-to-agol/index.html]: this sample is already present in `gh-pages` (via #290). it'd be a good idea to start separating out your work on a single sample in unique named branches for clarity.

**Comment by nhaney90** [comment]
Hi @jgravois, I made the requested changes

File: web-js/add-timestamp-to-wms-request/README.txt [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#Add Timestamp to WMS Requests
+
```
Comment (jgravois) [inline web-js/add-timestamp-to-wms-request/README.txt]: now you're proposing to merge in a `README.md` *and* a `README.txt`

**Comment by nhaney90** [comment]
Darn it, notepad++ always tries to save the readme file as .txt instead of .md

**Comment by jgravois** [comment]
nice work. squashed and merged.

@briantwatson can you sync the `master` branch manually now?

**Pull request merged**